### PR TITLE
Makefile: always use buildx

### PR DESCRIPTION
**Pull request by crazy-max**
follow-up #43529 (https://github.com/moby/moby/pull/43529/commits/86f9e1e1faacdcda255374ba64bf92ad05afe127)

Always use buildx for build commands.

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,17 +1,12 @@
...
-
-ifndef USE_BUILDX
-DOCKER_BUILDKIT := 1
-export DOCKER_BUILDKIT
+ifneq (, $(BUILDX_BIN))
+	BUILDX := $(BUILDX_BIN)
+else ifneq (, $(shell docker buildx version))
+	BUILDX := docker buildx
+else ifneq (, $(shell which buildx))
+	BUILDX := $(which buildx)
```
Comment (thaJeztah) [inline Makefile:9]: Wondering if we need this, can we just make `BUILDX` the way to override, and default it to `docker buildx` (or `docker`, once we have v23.0.0 I guess?)

File: Makefile [hunk 2]
```diff
@@ -214,11 +204,8 @@ build: shell_target := --target=dev
 else
 build: shell_target := --target=final
 endif
-ifdef USE_BUILDX
-build: buildx_load := --load
-endif
 build: buildx
-	$(BUILD_CMD) $(BUILD_OPTS) $(shell_target) $(buildx_load) $(BUILD_CROSS) -t "$(DOCKER_IMAGE)" .
+	$(BUILD_CMD) $(BUILD_OPTS) $(shell_target) --load $(BUILD_CROSS) -t "$(DOCKER_IMAGE)" .
```
Comment (thaJeztah) [inline Makefile:208]: Will the need for `--load` (and custom builders) go away once we have the containerd integration? Perhaps we need a variable to allow switching once that arrives.

File: Makefile [hunk 3]
```diff
@@ -271,12 +258,11 @@ swagger-docs: ## preview the API documentation
...
-buildx: bundles/buildx ## build buildx cli tool
-endif
-endif
-
-bundles/buildx: bundles ## build buildx CLI tool
+buildx: bundles/buildx ## download buildx CLI tool
+bundles/buildx:
 	curl -fsSL https://raw.githubusercontent.com/moby/buildkit/70deac12b5857a1aa4da65e90b262368e2f71500/hack/install-buildx | VERSION="$(BUILDX_VERSION)" BINDIR="$(@D)" bash
 	$@ version
+else
```
Comment (thaJeztah) [inline Makefile:266]: I'm a bit fuzzy on where we use `bundles/buildx`; is this on the host or inside the container? If it's on the host, can we assume `docker buildx` is pre-installed (requirement). And inside the dev container we can make sure it's installed. (We should have another look at updating the CLI, but possibly keep a second (older) cli around _only_ for the integration-cli tests for now)

File: Makefile (hunk 1 above)
Comment (crazy-max) [reply Makefile:9]: I guess we can yes as current docker cli package (20.10) install buildx plugin.

File: Makefile (hunk 2 above)
Comment (crazy-max) [reply Makefile:208]: Yes but the switch should be set in follow-up in https://github.com/rumpl/moby/ when rebased with this PR.

File: Makefile (hunk 3 above)
Comment (crazy-max) [reply Makefile:266]: On the host yes. Looking at what we said in https://github.com/moby/moby/pull/44533#discussion_r1032378391, I go ahead and remove this.

**Comment by thaJeztah** [comment]
Looking from my phone, but I think there's a left over `buildx` target in the makefile; https://github.com/moby/moby/blob/8c0c6994c7749c21d04afc0ecbf9722ac5b6c0ad/Makefile#L198

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**